crypto/cryptlib.c

```diff
@@ -396,15 +396,34 @@ void OPENSSL_showfatal(const char *fmta, ...)
 # endif
 }
 #else
+
+# ifdef __APPLE__
+#  include <Availability.h>
+#   if (defined(__MAC_10_11)  && __MAC_OS_X_VERSION_MIN_REQUIRED  >= __MAC_10_11) || \
+       (defined(__IPHONE_9_0) && __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_9_0)
+#    define HAVE_OS_CRASH
+extern void _os_crash(const char *);
```

Comment:
This appears to be an undocumented API. I'm not sure it is wise to use it?

Reply:
Hmmm, we could weak-link to it (which would mean we could check at runtime if the symbol is present or not). Alternately, we could use `__assert_rtn`, which is also undocumented but is called from the `assert()` macro and thus has frozen ABI, but that requires more structured input, and logs to stderr as well; see e.g.:
```
Assertion failed: (!"Voluntary assertion failure"), function main, file test/asserttest.c, line 14.
```
I suppose we could add `OpenSSL internal error: ` before the message with an asprintf?